pkg/controllers/datastoresyncer/node_handler.go

```diff
@@ -35,7 +37,21 @@ func (d *DatastoreSyncer) handleCreateOrUpdateNode(obj runtime.Object, numRequeu
 
 	globalIPOfNode := node.GetAnnotations()[constants.SmGlobalIP]
 
-	return d.updateLocalEndpointIfNecessary(globalIPOfNode)
+	// Validate that globalIPOfNode falls in the globalCIDR allocated to the cluster.
+	if globalIPOfNode != "" {
+		_, ipnet, err := net.ParseCIDR(d.localCluster.Spec.GlobalCIDR[0])
```

Comment:
Should we assume the array is  non-empty? Also it seems to assume there's only one entry -  what if there is more than one?

Reply:
>Should we assume the array is non-empty? 

This code runs only when Globalnet is enabled. 

>Also it seems to assume there's only one entry - what if there is more than one?

Currently we don't support multiple GlobalCIDRs and the implementation considers only one GlobalCIDR. You can look at the IPAM code for reference.